src/mixins/common-mixin.ts

```diff
@@ -68,4 +74,27 @@ export default class CommonMixin extends Vue {
     }
     return true
   }
+
+  /**
+   * Navigates to the dashboard page, optionally with a filing ID.
+   * @param identifier The identifier to include in the dashboard URL
+   * @param filingId The filing ID to include in the dashboard URL (optional)
+   */
+  protected navigateToDashboard (identifier: string, filingId?: number): void {
+    if (GetFeatureFlag('use-business-dashboard')) {
+      // Disable 'beforeunload' event
+      window.onbeforeunload = null
```

Comment:
Why is this needed if the FF is set but not if it isn't (below)?

I kinda don't like that we're changing a window handler here... I prefer to have the code that sets and the code that clears event handlers nearby each other -- kind of a symmetry,  you know?

Reply:
I noticed this when I click "Save and Resume" - without clearing the event handler, a dialog pops out saying "you are leaving this page, you may lost unsaved..."., because this page is navigating to an external URL. But actually the "Save and Resume" will save the information and this dialog is not needed.
However, when the feature flag is disabled, the Vue Router doesn't trigger it.

Reply:
Yes, I see.

In fact, this check for unsaved changes isn't even working correctly right now. Any exit from the filing view without first saving your data should trigger the browser alert, but if you start a filing and change some things and click the breadcrumb to return to the dashboard, you don't get the alert. This is because the user is not leaving the "window" (web app); they are only leaving one of the pages.

(Speaking of breadcrumbs, did you update it to navigate to the business dashboard?)

Let's not worry about this small bug right now.

However, I'm curious why you are getting this alert on save and resume. Theoretically, on successful save, the "haveChanges" flag should be cleared the the event handler should do nothing. Please test a little more to find which scenario is causing this and why. I don't really like the solution above.

As for the symmetry of setting and clearing event handlers, I still think this is important generally, but clearing is not so important when navigating to another UI because the window object gets destroyed and re-created.

Reply:
Thanks!
I fixed the problem, the onClickSaveResumeFinish () function didn't clear flag (this.haveChanges = false). I updated it.
I updated the navigate to the business dashboard breadcrumbs too.

Reply:
Excellent! Good fix.

The breadcrumbs look good, too.